provide.go

```diff
@@ -411,25 +441,31 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) (err error) {
 	oldProviders := make(map[key][]*constructorNode)
 	for k := range keys {
 		// Cache old providers before running cycle detection.
-		oldProviders[k] = c.providers[k]
-		c.providers[k] = append(c.providers[k], n)
+		oldProviders[k] = s.providers[k]
+		s.providers[k] = append(s.providers[k], n)
 	}
 
-	c.isVerifiedAcyclic = false
-	if !c.deferAcyclicVerification {
-		if ok, cycle := graph.IsAcyclic(c.gh); !ok {
-			// When a cycle is detected, recover the old providers to reset
-			// the providers map back to what it was before this node was
-			// introduced.
-			for k, ops := range oldProviders {
-				c.providers[k] = ops
+	for _, s := range allScopes {
+		s.isVerifiedAcyclic = false
+		if !s.deferAcyclicVerification {
+			if ok, cycle := graph.IsAcyclic(s.gh); !ok {
+				// When a cycle is detected, recover the old providers to reset
+				// the providers map back to what it was before this node was
+				// introduced.
+				for k, ops := range oldProviders {
+					s.providers[k] = ops
+				}
+
+				return errf("this function introduces a cycle", s.cycleDetectedError(cycle))
 			}
-
-			return errf("this function introduces a cycle", c.cycleDetectedError(cycle))
+			s.isVerifiedAcyclic = true
 		}
-		c.isVerifiedAcyclic = true
 	}
-	c.nodes = append(c.nodes, n)
+
+	// Before appending to the nodes, check if there are
+	// any child Scopes. If there are any, recurse down
+	// the descendents to provide the constructor to them.
+	s.nodes = append(s.nodes, n)
```

Comment:
Where is this check happening? I only see the append.

Reply:
sorry, that's an outdated comment. will remove